libcxx/test/std/experimental/reflection/member-classification.pass.cpp

```diff
@@ -359,6 +359,154 @@ static_assert(!is_explicit(^::));
+static_assert(is_noexcept(^S::noexcept_true_template_method<int>));
+static_assert(!is_noexcept(^S::noexcept_false_template_method<int>));
+static_assert(!is_noexcept(^S::not_noexcept_template_method<int>));
+
+// (no-)noexcept instantiated template method types
+static_assert(is_noexcept(type_of(^S::noexcept_template_method<int>)));
+static_assert(is_noexcept(type_of(^S::noexcept_true_template_method<int>)));
+static_assert(!is_noexcept(type_of(^S::noexcept_false_template_method<int>)));
+static_assert(!is_noexcept(type_of(^S::not_noexcept_template_method<int>)));
+
+// (no-)noexcept lambdas
+static_assert(is_noexcept(^noexcept_lambda));
+static_assert(!is_noexcept(^not_noexcept_lambda));
+
+// (no-)noexcept closure types
+static_assert(is_noexcept(type_of(^noexcept_lambda)));
+static_assert(!is_noexcept(type_of(^not_noexcept_lambda)));
+
+// (no-)noexcept function
+static_assert(is_noexcept(^noexcept_function));
+static_assert(is_noexcept(^noexcept_true_function));
+static_assert(!is_noexcept(^noexcept_false_function));
+static_assert(!is_noexcept(^not_noexcept_function));
+
+// (no-)noexcept function type
+static_assert(is_noexcept(type_of(^noexcept_function)));
+static_assert(is_noexcept(type_of(^noexcept_true_function)));
+static_assert(!is_noexcept(type_of(^noexcept_false_function)));
+static_assert(!is_noexcept(type_of(^not_noexcept_function)));
+
+// (no-)noexcept instantiated template functions
+static_assert(is_noexcept(^noexcept_template_function<int>));
+static_assert(is_noexcept(^noexcept_true_template_function<int>));
+static_assert(!is_noexcept(^noexcept_false_template_function<int>));
+static_assert(!is_noexcept(^not_noexcept_template_function<int>));
+
+// (no-)noexcept instantiated template function types
+static_assert(is_noexcept(type_of(^noexcept_template_function<int>)));
+static_assert(is_noexcept(type_of(^noexcept_true_template_function<int>)));
+static_assert(!is_noexcept(type_of(^noexcept_false_template_function<int>)));
+static_assert(!is_noexcept(type_of(^not_noexcept_template_function<int>)));
+
+// The rest (should all be false regardless of noexcept specifier)
+// (no-)noexcept non-instantiated template methods
+static_assert(!is_noexcept(^S::noexcept_template_method));
+static_assert(!is_noexcept(^S::not_noexcept_template_method));
+
+// (no-)noexcept generic closure types
+static_assert(!is_noexcept(^noexcept_generic_lambda));
+static_assert(!is_noexcept(^not_noexcept_generic_lambda));
+static_assert(!is_noexcept(type_of(^noexcept_generic_lambda)));
+static_assert(!is_noexcept(type_of(^not_noexcept_generic_lambda)));
+
+// (no-)noexcept non-instantiated template functions
+static_assert(!is_noexcept(^noexcept_template_function));
+static_assert(!is_noexcept(^not_noexcept_template_function));
+
+// Random non callable types
+auto x = 1;
+auto s = S();
+
+static_assert(!is_noexcept(^x));
+static_assert(!is_noexcept(type_of(^x)));
+static_assert(!is_noexcept(^s));
+static_assert(!is_noexcept(type_of(^s)));
```

Comment:
Instead of these checks, would you mind adding a
```cpp
static_assert(!is_noexcept(^whatever));
```

for all of the different kinds of things at the top of this file that can't be noexcept (e.g., namespaces, templates, etc)?

Reply:
Should it be at the top of the file, or at the top of is_noexcept test block? The rest of the tests seem to be grouped together. I'm ok with separating, just wanted to clarify I got what you mean correctly

Reply:
Ah, nevermind; I was thinking of a different file. Please disregard :)